components/telemetry-operator/controller/tracepipeline/config_builder.go

```diff
@@ -0,0 +1,223 @@
+package tracepipeline
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+)
+
+type TlsConfig struct {
+	Insecure bool `yaml:"insecure"`
+}
+
+type OtlpExporterConfig struct {
+	Endpoint       string         `yaml:"endpoint,omitempty"`
+	Headers        map[string]any `yaml:"headers,omitempty"`
+	Tls            TlsConfig      `yaml:"tls,omitempty"`
+	SendingQueue   map[string]any `yaml:"sending_queue,omitempty"`
+	RetryOnFailure map[string]any `yaml:"retry_on_failure,omitempty"`
+}
+
+type ExporterConfig struct {
+	Otlp     OtlpExporterConfig `yaml:"otlp,omitempty"`
+	OtlpHttp OtlpExporterConfig `yaml:"otlphttp,omitempty"`
+}
+
+type ReceiverConfig struct {
+	OpenCensus map[string]any `yaml:"opencensus"`
+	Otlp       map[string]any `yaml:"otlp"`
+}
+
+type ProcessorsConfig struct {
+	Batch         map[string]any `yaml:"batch,omitempty"`
+	MemoryLimiter map[string]any `yaml:"memory_limiter,omitempty"`
+	K8sAttributes map[string]any `yaml:"k8sattributes,omitempty"`
+	Resource      map[string]any `yaml:"resource,omitempty"`
+}
+
+type PipelineConfig struct {
+	Receivers  []string `yaml:"receivers"`
+	Processors []string `yaml:"processors"`
+	Exporters  []string `yaml:"exporters"`
+}
+
+type PipelinesConfig struct {
+	Traces PipelineConfig `yaml:"traces"`
+}
+
+type OtlpServiceConfig struct {
+	Pipelines  PipelinesConfig `yaml:"pipelines,omitempty"`
+	Telemetry  map[string]any  `yaml:"telemetry,omitempty"`
+	Extensions []string        `yaml:"extensions,omitempty"`
+}
+
+type ExtensionsConfig struct {
+	HealthCheck map[string]any `yaml:"health_check"`
+}
+
+type OtelCollectorConfig struct {
+	Receivers  ReceiverConfig    `yaml:"receivers"`
+	Exporters  ExporterConfig    `yaml:"exporters"`
+	Processors ProcessorsConfig  `yaml:"processors"`
+	Extensions ExtensionsConfig  `yaml:"extensions"`
+	Service    OtlpServiceConfig `yaml:"service"`
+}
+
+func makeReceiverConfig() ReceiverConfig {
+	return ReceiverConfig{
+		OpenCensus: map[string]any{},
+		Otlp: map[string]any{
```

Comment:
Would it make sense to have a type, for instance, for receivers, exporters, etc? Is there any benefit in having it untyped (as a map[string]any)?

Reply:
Ok, let's leave it like this for now.

Reply:
Introduced types, except for the expected empty maps.